### ISSUE TITLE
vscocde got killed after updating to OSX 12.3

### ISSUE THREAD

**Issue by victoryang00** [microsoft/vscode#145417]
VSCode Version: Version: 1.65.2 Arm64
Local OS Version: macOS 12.3 (21E230)

Steps to Reproduce:

1. Got killed after updating to the latest OSX

```bash
/opt/homebrew/Library/Taps/riscv/homebrew-riscv on  master via 💎 v2.6.8 took 1m7s
❯ "/Applications/Visual Studio Code.app/Contents/MacOS/Electron"                                                    (base)
[main 2022-03-18T19:17:53.220Z] window#load: attempt to load window (id: 1)
[main 2022-03-18T19:17:53.260Z] window#load: attempt to load window (id: 2)
[main 2022-03-18T19:17:53.310Z] window#load: attempt to load window (id: 3)
[main 2022-03-18T19:17:53.336Z] window#load: attempt to load window (id: 4)
[main 2022-03-18T19:17:53.340Z] update#setState idle
[0319/031753.484310:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031753.617915:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[0319/031753.625279:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031753.757182:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031753.767493:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[0319/031754.122333:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031754.263243:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031754.272676:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[0319/031754.673842:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031754.803218:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031754.814964:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[0319/031755.187888:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031755.315375:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031755.324837:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[0319/031755.647832:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031755.775893:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031755.785122:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[0319/031756.102023:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031756.231094:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031756.241626:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[0319/031756.542702:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031756.670589:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031756.680574:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[main 2022-03-18T19:17:58.431Z] ExtensionHostStarterWorker created
[54471:0319/031758.432236:ERROR:network_service_instance_impl.cc(331)] Network service crashed, restarting service.
[0319/031758.927169:ERROR:appcenter_api.cc(341)] {"status":"AllLogsThrottled","validDiagnosticsIds":[],"throttledDiagnosticsIds":["22eafe63-0152-4c6c-a259-8223ea457da5","f3d669ee-ef68-4cec-a42b-13683a1c2a5e"],"correlationId":"5e02116a-ba47-4b08-bef2-c4407759d406"}
[0319/031759.155528:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031759.282423:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[main 2022-03-18T19:17:59.293Z] CodeWindow: renderer process crashed (reason: killed, code: 9)
[0319/031759.432852:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031759.563063:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[0319/031759.572348:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[main 2022-03-18T19:17:59.595Z] CodeWindow: renderer process crashed (reason: killed, code: 9)
[0319/031759.706479:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[0319/031759.713999:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031759.843283:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[0319/031759.852016:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[main 2022-03-18T19:17:59.884Z] CodeWindow: renderer process crashed (reason: killed, code: 9)
[0319/031759.980919:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[0319/031759.989777:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031800.121216:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031800.131720:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[main 2022-03-18T19:18:00.169Z] CodeWindow: renderer process crashed (reason: killed, code: 9)
[54471:0319/031800.502838:ERROR:network_service_instance_impl.cc(331)] Network service crashed, restarting service.
[0319/031800.819661:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[0319/031800.944601:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[0319/031800.955956:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
[54471:0319/031800.957136:ERROR:network_service_instance_impl.cc(331)] Network service crashed, restarting service.
[0319/031801.086172:WARNING:exception_snapshot_mac.cc(70)] exception EXC_GUARD invalid in EXC_CRASH
[54471:0319/031801.095335:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=9
[54471:0319/031801.095375:FATAL:gpu_data_manager_impl_private.cc(415)] GPU process isn't usable. Goodbye.
[0319/031801.373040:WARNING:crash_report_exception_handler.cc(240)] UniversalExceptionRaise: (os/kern) failure (5)
[0319/031801.374148:WARNING:exception_types.cc(171)] EXC_CRASH should not contain exception 12
fish: Unknown command: fish_job_summary
in event handler: handler for generic event 'fish_job_summary'
```

**Comment by victoryang00** [comment]
My machine is M1 Max

**Comment by Gamizard** [comment]
I'm getting this too, using a M1 MBP on macOS 12.3.

**Comment by victoryang00** [comment]
I fixed it by resetting the `csrutils enable` and `csrutils disable`. OSX seems to have some OS implementation relying on the full security system.

**Comment by alexdima** [comment]
FYI @deepak1556

**Comment by CCBlueX666** [comment]
I'm getting this too, reopen the sip. It will work

**Comment by a4840639** [comment]
@victoryang00 Did you ended up disabling SIP? I do not think it is a solution but merely a workaround. It is probably 100% Apple's fault to ship a broken OS, though

**Comment by JacobSyndeo** [comment]
![image](https://user-images.githubusercontent.com/1004637/162833029-77c6327f-762b-4cd9-b061-b55affa90284.png)
As it turns out, according to `csrutil`, my machine was in some weird limbo state, and _that_ was in turn causing anything Chromium (Teams, Slack, etc) to either not open, crash upon launch, or show up with completely empty/blank windows.

Booting to the recovery environment and running `csrutil enable` fixed it for me. Presumably, `csrutil disable` would have worked as well.

Thanks to @victoryang00 for mentioning SIP; I wasn't aware things could be affected by it to this degree!

**Comment by KashKask** [comment]
Thank you @victoryang00 and @JacobSyndeo, setting `csrutil enable` was exactly what my MacBook needed. I had this crashing issue for a while after upgrading to the MacOS 12.3... I checked my current status with `csrutil status` and saw that mine was also in a custom state. 
I tried to reinstall the OS (while keeping the files), but with no success. I was on my way to reimage the unit and install fresh OS. Also, I thought that it had something to do with resetting the privacy database `sudo tccutil reset ALL` but it didn't fix the issue for me...
